### PR TITLE
Added some testing jobs and descriptions

### PR DESCRIPTION
**Pull request by jjasghar**
- Added a description for `rake test`
- moved the order of the dependancies
- added rubocop

Now if we decide to use rubocop, running without a `.rubocop.yml` it
comes back with

```
773 files inspected, 12455 offenses detected
```

I suggest we decide on a `.rubocop.yml` before we start doing anything
large with this Gem.

### CODE REVIEW

**Comment by Ladas** [comment]
the one we use in ManageIQ, https://github.com/ManageIQ/manageiq/blob/master/.rubocop.yml for inspiration :-)

though we take some of them more as recommendations, rewriting everything to 25 methods with low complexity can be hard sometimes :-)

**Comment by jjasghar** [comment]
Just to update, after running that version @Ladas it came back with:

```
773 files inspected, 6592 offenses detected
```

I did have to updated a few of the Cops though:
```
Error: The `Style/TrailingComma` cop no longer exists. Please use `Style/TrailingCommaInLiteral` and/or `Style/TrailingCommaInArguments` instead.
Error: The `Style/SingleSpaceBeforeFirstArg` cop has been renamed to `Style/SpaceBeforeFirstArg.
Error: obsolete parameter RunRailsCops (for AllCops)
```

**Comment by Ladas** [comment]
If nobody is against, I would merge it as a first version, once we remove the ManageIQ specific checks. We can tweak the config later, if it will become more obtrusive then helpful. ;-)

**Comment by jjasghar** [comment]
:+1:

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+StringLiteralsInInterpolation:
+  Enabled: false
+TrailingCommaInLiteral:
+  Enabled: false
+WhileUntilModifier:
+  Enabled: false
+WordArray:
+  AutoCorrect: false
+
+ClassAndModuleCamelCase:
```
Comment (Ladas) [inline .rubocop.yml]: Seems like you forgot L95-L101

hm I wonder why it doesn't show my comments in the code

Comment (Ladas) [inline .rubocop.yml]: Ah, I probably commented the commit directly

**Comment by Ladas** [comment]
:+1: cool thank you

**Pull request merged**